src/app/services/data.service.ts

```diff
@@ -1,10 +1,20 @@
 import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { ApiResponse, Character } from '../model/character.model'; 
 
-@Injectable({providedIn: 'root'})
-
+@Injectable({
+  providedIn: 'root'
+})
 export class DataService {
-    constructor() { }
+  private apiUrl = 'https://rickandmortyapi.com/api/character';
+
+  constructor(private http: HttpClient) {}
 
-    // Implement a method to retrieve data from the following URL: https://rickandmortyapi.com/api/character
-   
-}
\ No newline at end of file
+  getCharacters(): Observable<Character[]> {
+    return this.http.get<ApiResponse>(`${this.apiUrl}?page=1`).pipe(
+      map(response => response.results.slice(0, 20))  
```

Comment:
It's always good design to add catchError operators for any http calls because of the uncertainty of server responses.